src/lib/browser/Browser.ts

```diff
@@ -24,50 +30,82 @@ export class Browser {
     return this.#browser;
   }
 
-  async getCurrentConcurrency(): Promise<number> {
-    return (await this.#browser!.pages()).length;
-  }
-
   /**
-   * Create a puppeteer instance.
+   * Create a Playwright instance.
    */
   async create(): Promise<void> {
-    console.info(`Browser ${this.#id} creating...`);
+    log.info('Creating...', { id: this.#id });
 
     const env: { [s: string]: string } = {};
     if (process.env.DISPLAY) {
       env.DISPLAY = process.env.DISPLAY;
     }
 
     let start = Date.now();
-    const browser = await puppeteer.launch({
+    const browser = await chromium.launch({
       headless: true,
       env,
-      executablePath: await getChromiumExecutablePath(),
-      defaultViewport: {
-        width: WIDTH,
-        height: HEIGHT,
-      },
       handleSIGINT: false,
+      handleSIGHUP: false,
       handleSIGTERM: false,
-      pipe: true,
       args: flags,
     });
+    this.#browser = browser;
     stats.timing('renderscript.create', Date.now() - start);
 
     // Try to load a test page first
     start = Date.now();
-    const testPage = await browser.newPage();
-    await testPage.goto('about://settings', { waitUntil: 'networkidle0' });
+    const context = await this.getNewContext({});
+    const testPage = await context.newPage();
+    await testPage.goto('about://settings', {
+      waitUntil: 'networkidle',
+      timeout: 2000,
+    });
     stats.timing('renderscript.page.initial', Date.now() - start);
-    testPage.close();
+    await testPage.close();
+    await context.close();
 
-    this.#browser = browser;
     this.#ready = true;
-    console.log(`Browser is ready`);
+    log.info('Ready', { id: this.#id });
   }
 
   async stop(): Promise<void> {
     await this.#browser?.close();
   }
+
+  getCurrentConcurrency(): number {
+    if (!this.#browser) {
+      return 0;
+    }
+
+    return this.#browser.contexts().reduce((i, ctx) => {
+      return i + ctx.pages().length;
+    }, 0);
+  }
+
+  async getNewContext(opts: BrowserContextOptions): Promise<BrowserContext> {
+    const start = Date.now();
+    const ctx = await this.#browser!.newContext({
+      acceptDownloads: false,
+      bypassCSP: false,
+      hasTouch: false,
+      isMobile: false,
+      javaScriptEnabled: true,
+      locale: 'en-GB',
+      timezoneId: 'Europe/Paris',
+      offline: false,
+      permissions: [],
+      userAgent: 'Algolia Crawler Renderscript',
+      viewport: { height: HEIGHT, width: WIDTH },
+      extraHTTPHeaders: {
+        'Accept-Encoding': 'gzip, deflate',
+        Accept:
+          'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
```

Comment:
What is it useful for?

Reply:
It's the standard headers sent by chrome, not sure what is was so I specified here. 
But now that I re-read it I'll remove the image part since we block images anywway.